### PR TITLE
Add nested loops

### LINKED ISSUE #17208: The documentation doesn't mention any examples for nested loops
The challenge requires to use nested loops, which are in fact mentioned, but there just aren't any examples on how they work

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 77dec100-463d-f684-03b6-25f511d1d18f
* Version Independent ID: 27921da6-71b0-6dfc-2824-c84844152d8e
* Content: [Branches and loops - Introduction to C# interactive tutorial](https://docs.microsoft.com/en-us/dotnet/csharp/tutorials/intro-to-csharp/branches-and-loops?tutorial-step=7#feedback)
* Content Source: [docs/csharp/tutorials/intro-to-csharp/branches-and-loops.yml](https://github.com/dotnet/docs/blob/master/docs/csharp/tutorials/intro-to-csharp/branches-and-loops.yml)
* Product: **dotnet-csharp**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by BillWagner**
The beginner loops tutorial does not contain any examples of nested loops.

Add a beginner example to both the interactive and local versions.

Fixes #17208

[internal review link - interactive](https://review.docs.microsoft.com/en-us/dotnet/csharp/tutorials/intro-to-csharp/branches-and-loops?branch=pr-en-us-18039&tutorial-step=5)
[Internal review link - local](https://review.docs.microsoft.com/en-us/dotnet/csharp/tutorials/intro-to-csharp/branches-and-loops-local?branch=pr-en-us-18039#created-nested-loops)

### CODE REVIEW

File: docs/csharp/tutorials/intro-to-csharp/branches-and-loops-local.md [hunk 1]
```diff
@@ -276,6 +276,42 @@ Experiment with these yourself. Try each of the following:
...
+for (int row = 1; row < 11; row++)
+{
+    for (char column = 'a'; column < 'k'; column++)
+    {
+        Console.WriteLine($"The cell is ({row}, {column})");
+    }
+}
+```
+
+You can see that the outer out loop increments once for each iteration of the inner loop. Reverse the row and column nesting, and see the changes for yourself.
```
Comment (pkulikov) [inline docs/csharp/tutorials/intro-to-csharp/branches-and-loops-local.md:313]: > the outer out loop increments once for each iteration of the inner loop

That doesn't sound correct. I've read it as `row` is incremented once for each iteration of the inner loop, that is, for each increment of `column`. 

Did you want to say:
```suggestion
You can see that the outer loop increments once for each full run of the inner loop. Reverse the row and column nesting, and see the changes for yourself.
```

File: docs/csharp/tutorials/intro-to-csharp/branches-and-loops.yml [hunk 2]
```diff
@@ -237,6 +237,46 @@ items:
...
+    {
+      for (char column = 'a'; column < 'k'; column++)
+      {
+        Console.WriteLine($"The cell is ({row}, {column})");
+      }
+    }
+    ```
+
+    You can see that the outer out loop increments once for each iteration of the
+    inner loop. Reverse the row and column nesting, and see the changes for yourself.
```
Comment (pkulikov) [inline docs/csharp/tutorials/intro-to-csharp/branches-and-loops.yml:278]:
```suggestion
    You can see that the outer loop increments once for each full run of the
    inner loop. Reverse the row and column nesting, and see the changes for yourself.
```

File: docs/csharp/tutorials/intro-to-csharp/branches-and-loops-local.md [hunk 3]
```diff
@@ -276,6 +276,42 @@ Experiment with these yourself. Try each of the following:
 When you're done, let's move on to write some code yourself to
 use what you've learned.
 
+## Created nested loops
+
+A `while`, `do` or `for` loop can be nested inside another loop to create a matrix using the combination of each item in the outer loop with each item in the inner loop. Let's do that to build a set of alpha-numeric pairs to represent rows and columns.
```
Comment (tdykstra) [inline docs/csharp/tutorials/intro-to-csharp/branches-and-loops-local.md:281]:
```suggestion
A `while`, `do` or `for` loop can be nested inside another loop to create a matrix using the combination of each item in the outer loop with each item in the inner loop. Let's do that to build a set of alphanumeric pairs to represent rows and columns.
```

File: docs/csharp/tutorials/intro-to-csharp/branches-and-loops.yml [hunk 4]
```diff
@@ -237,6 +237,46 @@ items:
     > [!NOTE]
     > This online coding experience is in preview mode. If you encounter problems, please report them [on the dotnet/try repo](https://github.com/dotnet/try/issues).
 
+- title: Created nested loops
+  durationInMinutes: 10
+  content: |
+    A `while`, `do` or `for` loop can be nested inside another loop to create a matrix 
+    using the combination of each item in the outer loop with each item in the inner
+    loop. Let's do that to build a set of alpha-numeric pairs to represent rows and columns.
```
Comment (tdykstra) [inline docs/csharp/tutorials/intro-to-csharp/branches-and-loops.yml:245]:
```suggestion
    loop. Let's do that to build a set of alphanumeric pairs to represent rows and columns.
```

**Review by tdykstra — APPROVED** [review]
Looks great, just a couple minor suggestions.  What is the YAML file used for?

**Comment by BillWagner** [comment]
@tdykstra 

>  What is the YAML file used for?

That's how we build the step-by-step interactive tutorials.

**Pull request merged**